File: src/backend/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ReactCore.Backend.Models;

public class Product
{
    public int Id { get; set; }

    [Required]
    [MaxLength(255)]
    public string Name { get; set; } = string.Empty;

    public string? Description { get; set; }

    [Column(TypeName = "decimal(10,2)")]
    public decimal Price { get; set; }

    [Required]
    [MaxLength(100)]
    public string Category { get; set; } = string.Empty;

    [MaxLength(500)]
    public string? ImageUrl { get; set; }

    public string? Specifications { get; set; }

    public int StockQuantity { get; set; }

    public int ReorderLevel { get; set; } = 5;

    public bool IsDeleted { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public List<ProductImage> Images { get; set; } = new();

    public List<OrderItem> OrderItems { get; set; } = new();

    public List<InventoryAudit> InventoryAudits { get; set; } = new();
}

File: src/frontend/src/types/admin/Product.ts
export interface AdminProductListItem {
  id: number;
  name: string;
  price: number;
  category: string;
  stockQuantity: number;
  reorderLevel: number;
  isDeleted: boolean;
  updatedAt: string;
}

export interface AdminProductDetail {
  id: number;
  name: string;
  description?: string | null;
  price: number;
  category: string;
  imageUrl?: string | null;
  imageUrls: string[];
  specifications: Record<string, unknown>;
  stockQuantity: number;
  reorderLevel: number;
  isDeleted: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface AdminProductUpsertRequest {
  name: string;
  description?: string | null;
  price: number;
  category: string;
  stockQuantity: number;
  reorderLevel?: number;
  specifications?: Record<string, unknown>;
  imageUrl?: string | null;
}

export interface AdminProductCreateResponse {
  id: number;
  name: string;
  price: number;
  category: string;
  stockQuantity: number;
  createdAt: string;
}

export interface AdminProductUpdateResponse {
  success: boolean;
  productId: number;
  updatedAt: string;
}
